enterprise/models.py

```diff
@@ -1028,3 +1030,143 @@ def __repr__(self):
+    """
+
+    FREQUENCY_TYPE_DAILY = 'daily'
+    FREQUENCY_TYPE_MONTHLY = 'monthly'
+    FREQUENCY_TYPE_WEEKLY = 'weekly'
+
+    FREQUENCY_CHOICES = (
+        (FREQUENCY_TYPE_DAILY, FREQUENCY_TYPE_DAILY),
+        (FREQUENCY_TYPE_MONTHLY, FREQUENCY_TYPE_MONTHLY),
+        (FREQUENCY_TYPE_WEEKLY, FREQUENCY_TYPE_WEEKLY),
+    )
+
+    DELIVERY_METHOD_EMAIL = 'email'
+    # We are adding this field preemptively as we plan to support FTP, but don't have the details yet.
+    DELIVERY_METHOD_FTP = 'ftp'
+
+    DELIVERY_METHOD_CHOICES = (
+        (DELIVERY_METHOD_EMAIL, DELIVERY_METHOD_EMAIL),
+        (DELIVERY_METHOD_FTP, DELIVERY_METHOD_FTP),
+    )
+
+    DAYS_OF_WEEK = (
+        (0, 'Monday'),
+        (1, 'Tuesday'),
+        (2, 'Wednesday'),
+        (3, 'Thursday'),
+        (4, 'Friday'),
+        (5, 'Saturday'),
+        (6, 'Sunday'),
+    )
+
+    enterprise_customer = models.OneToOneField(
+        EnterpriseCustomer, blank=False, null=False, verbose_name=_("Enterprise Customer")
+    )
+    active = models.BooleanField(blank=False, null=False, verbose_name=_("Active"))
+    delivery_method = models.CharField(
+        max_length=20,
+        choices=DELIVERY_METHOD_CHOICES,
+        blank=False,
+        default=DELIVERY_METHOD_EMAIL,
+        verbose_name=_("Delivery Method"),
+        help_text=_("The method in which the data should be sent.")
+    )
+    email = models.EmailField(blank=False, null=False, verbose_name=_("Email"))
+    frequency = models.CharField(
+        max_length=20,
+        choices=FREQUENCY_CHOICES,
+        blank=False,
+        default=FREQUENCY_TYPE_MONTHLY,
+        verbose_name=_("Frequency"),
+        help_text=_("The frequency interval (daily, weekly, or monthly) that the report should be sent."),
+    )
+    day_of_month = models.SmallIntegerField(
+        blank=True,
+        null=True,
+        verbose_name=_("Day of Month"),
+        help_text=_("The day of the month to send the report. "
+                    "This field is required and only valid when the frequency is monthly."),
+        validators=[MinValueValidator(1), MaxValueValidator(31)]
+    )
+    day_of_week = models.SmallIntegerField(
+        blank=True,
+        null=True,
+        choices=DAYS_OF_WEEK,
+        verbose_name=_("Day of Week"),
+        help_text=_("The day of the week to send the report. "
+                    "This field is required and only valid when the frequency is weekly."),
+    )
+    hour_of_day = models.SmallIntegerField(
+        blank=False,
+        null=False,
+        verbose_name=_("Hour of Day"),
+        help_text=_("The hour of the day to send the report, in Eastern Standard Time (EST). "
+                    "This is required for all frequency settings."),
+        validators=[MinValueValidator(0), MaxValueValidator(23)]
+    )
+    iv = models.BinaryField(   # pylint: disable=invalid-name
```

Comment:
What does this do?

Reply:
Nevermind. I get it. It would be good to add `help_text` or a comment explaining what this field is used for. Also, I would go with `password_encryption_initialization_vector` as a field name.